src/test/java/uk/gov/digital/ho/hocs/client/audit/AuditClientTest.java

```diff
@@ -46,62 +57,63 @@
     private final ClientContext clientContext = new ClientContext();
 
     @Captor
-    ArgumentCaptor<HashMap<String, Object>> headerCaptor;
-    @Captor
-    ArgumentCaptor<String> jsonMessage;
+    ArgumentCaptor<PublishRequest> snsPublishedMessage;
+
+    private ResultCaptor<PublishResult> snsPublishedResponse;
 
     @Before
     public void setUp() {
-        SNSTopicPrefix topicPrefix = new LocalStackConfiguration().topicPrefix();
-        AuditTopicBuilder auditTopicBuilder = new AuditTopicBuilder(topicPrefix, topicName);
-        topic = auditTopicBuilder.getTopic();
+        snsPublishedResponse = new ResultCaptor<>();
+        doAnswer(snsPublishedResponse).when(amazonSNS).publish(any());
+
         clientContext.setContext(userId, groups, team, correlationId);
-        auditClient = new AuditClient(
-                auditTopicBuilder,
+
+
+        this.auditClient = new AuditClient(
                 raisingService,
                 namespace,
                 new SpringConfiguration().objectMapper(),
                 clientContext,
-                producerTemplate);
+                amazonSNS,
+                awsSnsProperties);
     }
 
     @Test
     public void shouldWriteAuditWithoutPayload() {
-
         auditClient.audit(EventType.CREATOR_CASE_CREATED, caseId, stageId);
-        verify(producerTemplate).sendBodyAndHeaders(eq(topic), jsonMessage.capture(), headerCaptor.capture());
-        ReadContext ctx = JsonPath.parse(jsonMessage.getValue());
 
+        assertNotNull(snsPublishedResponse.getResult().getMessageId());
+        verify(amazonSNS).publish(snsPublishedMessage.capture());
+        ReadContext ctx = JsonPath.parse(snsPublishedMessage.getValue().getMessage());
```

Comment:
Thankyou for leaving this for another PR 👍

Reply:
You're more than welcome...